plugin/src/main/java/com/google/errorprone/xplat/checker/J2objcMethodName.java

```diff
@@ -0,0 +1,351 @@
+
+  private Description genDescription(String message, MethodTree tree) {
+    return buildDescription(tree)
+        .setMessage(message)
+        .addFix(SuggestedFix.builder()
+            .addImport("com.google.j2objc.annotations.ObjectiveCName")
+            .prefixWith(tree, String.format("@ObjectiveCName(\"%s\")\n",
+                getMethodName(tree.getName().toString())))
+            .build())
+        .build();
+  }
+
+  private String localClassLookupName(String enclosingClass, String outermostClass) {
+    return enclosingClass.substring(outermostClass.lastIndexOf(".") + 1);
+  }
+
+  private String localClassNameMangle(String enclosingClass, String outermostClass) {
+    StringBuilder newName = new StringBuilder();
+    String outermost = outermostClass.substring(outermostClass.lastIndexOf(".") + 1);
+
+    if (foundObjcClassNames.containsKey(outermost)) {
+      newName.append(foundObjcClassNames.get(outermost));
+    } else if (packageAnnotation != null) {
+      newName.append(packageAnnotation);
+      newName.append(outermost);
+    } else {
+      Iterable<String> packageParts = Splitter.on('.').split(outermostClass);
+
+      for (String part : packageParts) {
+        newName.append(part.substring(0, 1).toUpperCase());
+        newName.append(part.substring(1));
+      }
+    }
+
+    if (outermostClass.length() < enclosingClass.length()) {
+      Iterable<String> remainingParts = Splitter.on('.')
+          .split(enclosingClass.substring(outermostClass.length() + 1));
+      StringBuilder lookupName = new StringBuilder(outermost);
+
+      for (String part : remainingParts) {
+        newName.append("_");
+        lookupName.append(".");
+        lookupName.append(part);
+
+        if (foundObjcClassNames.containsKey(lookupName.toString())) {
+          newName = new StringBuilder(foundObjcClassNames.get(lookupName.toString()));
+        } else {
+          newName.append(part.substring(0, 1).toUpperCase());
+          newName.append(part.substring(1));
+        }
+      }
+    }
+    return newName.toString();
+  }
+
+  private String externalTypeNameMangle(String enclosingClass, String outermostClass) {
+    StringBuilder newName = new StringBuilder();
+    Iterable<String> packageParts = Splitter.on('.').split(outermostClass);
+
+    for (String part : packageParts) {
+      newName.append(part.substring(0, 1).toUpperCase());
+      newName.append(part.substring(1));
+    }
+
+    if (outermostClass.length() < enclosingClass.length()) {
+      Iterable<String> remainingParts = Splitter.on('.')
+          .split(enclosingClass.substring(outermostClass.length() + 1));
+
+      for (String part : remainingParts) {
+        newName.append("_");
+        newName.append(part.substring(0, 1).toUpperCase());
+        newName.append(part.substring(1));
+      }
+    }
+    return newName.toString();
+  }
+
+  private String parameterNameMangle(String name, List<? extends VariableTree> paramList) {
```

Comment:
Add a Javadoc here with an example. Same for the rest of the helper private methods below.

Reply:
Done